### **Clase 1 - Estadística Univariada**

Para instalar la libreria **`pandas`** en **`Jupyter`**:
- !python -m pip install pandas

Para instalar la libreria **`pandas`** en **`VSCode`**:
- pip install pandas

Para instalar la libreria **`scipy`** en **`VSCode`**:
- pip install scipy

Para instalar la libreria **`seaborn`** en **`VSCode`**:
- pip install seaborn

Para instalar la libreria **`plotly`** en **`VSCode`**:
- pip install plotly

#### **`Importar Librerias`**

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy as sp
import seaborn as sns
import plotly

#### **`Revisar versión Librerias`**

In [8]:
import pandas 
pandas.__version__

'1.4.1'

In [9]:
import numpy 
numpy.__version__

'1.26.2'

In [7]:
import scipy 
scipy.__version__

'1.11.4'

In [10]:
import matplotlib 
matplotlib.__version__

'3.5.0'

#### **`Listar archivos que se encuentran en el mismo directorio que el actual archivo notebook`**

In [2]:
import os

os.listdir()

['clase_1_-_estadistica_univariada.ipynb', 'flights.csv']

#### **`Importar un archivo CSV en un Dataframe`**

In [3]:
dt = pd.read_csv('flights.csv')

dt

year      month  passengers
0    1949    January         112
1    1949   February         118
2    1949      March         132
3    1949      April         129
4    1949        May         121
..    ...        ...         ...
139  1960     August         606
140  1960  September         508
141  1960    October         461
142  1960   November         390
143  1960   December         432

[144 rows x 3 columns]

In [7]:
type(dt)

pandas.core.frame.DataFrame

In [5]:
df = pd.read_csv('datos_persona.csv')

df

indice sexo  altura  peso     nombre
0       0    F    1.64    68    Javiera
1       1    M    1.53    43       Jose
2       2    M    1.87    90      Tomas
3       3    F    1.67    95      Maria
4       4    M    2.01   100       Jose
5       5    F    1.45    50  Magdalena
6       6    F    1.67    67   Trinidad
7       7    M    1.93   102    Gonzalo
8       8    M    1.72    76      David
9       9    M    1.64    68     Javier

#### **`Seleccionar una columna de un Dataframe`**

In [8]:
dt['month']

0        January
1       February
2          March
3          April
4            May
         ...    
139       August
140    September
141      October
142     November
143     December
Name: month, Length: 144, dtype: object

In [9]:
type(dt['month'])

pandas.core.series.Series

#### **`Escoger primeras filas de un Dataframe`**

**`head()`** por defecto escoge los 5 primeros registros

In [10]:
dt.head()

year     month  passengers
0  1949   January         112
1  1949  February         118
2  1949     March         132
3  1949     April         129
4  1949       May         121

In [11]:
dt.head(4)

year     month  passengers
0  1949   January         112
1  1949  February         118
2  1949     March         132
3  1949     April         129

In [12]:
dt.head(6)

year     month  passengers
0  1949   January         112
1  1949  February         118
2  1949     March         132
3  1949     April         129
4  1949       May         121
5  1949      June         135

#### **`Escoger últimas filas de un Dataframe`**

**`tail()`** por defecto escoge los 5 primeros registros

In [13]:
dt.tail()

year      month  passengers
139  1960     August         606
140  1960  September         508
141  1960    October         461
142  1960   November         390
143  1960   December         432

In [14]:
dt.tail(3)

year     month  passengers
141  1960   October         461
142  1960  November         390
143  1960  December         432

#### **`Dimensión del Dataframe`**

Nos devuelve una tupla:
- 144 registros
- 3 columnas

In [16]:
dt.shape

(144, 3)

#### **`Estadistica descriptiva - Calcular la MEDIA`**

Utilizando PANDAS

In [4]:
dt['passengers'].mean()

280.2986111111111

Utilizando NUMPY

In [5]:
np.mean(dt['passengers'])

280.2986111111111

#### **`Estadistica descriptiva - Calcular la MODA`**

Utilizando SCIPY

In [12]:
import scipy.stats as stats
stats.mode(dt['passengers'])

ModeResult(mode=229, count=3)

#### **`Estadistica descriptiva - Calcular la MEDIANA`**

In [13]:
a = [1,2,3,4,5]
np.mean(a), np.median(a)

(3.0, 3.0)

In [14]:
a = [1,2,3,4,100]
np.mean(a), np.median(a)

(22.0, 3.0)

In [18]:
a = [1,2,3,4,5,10]
np.mean(a), np.median(a)

(4.166666666666667, 3.5)

#### **`Estadistica descriptiva - Calcular la VARIANZA`**

Utilizando Pandas

In [19]:
dt['passengers'].var()

14391.917200854701

Utiliando Numpy

In [20]:
np.var(dt['passengers'])

14291.97333140432

Ambos resultados son distintos. Esto porque en NUMPY utiliza por defecto el argumento grados de libertad igual a 0 y en Pandas este argumento es igual a 1. 

<center><img src="https://i.postimg.cc/QdMT5SGy/dl3.png"></center>
<center><img src="https://i.postimg.cc/FHnJxchs/dl4.png"></center>

Utilizando Pandas

Por tanto, en Panda debemos igualarlo a 0:

In [4]:
dt['passengers'].var(ddof=0)

14291.97333140432

Para que quede mas claro esto de los **`GRADOS DE LIBERTAD`**

Siendo esta la formula de la **`VARIANZA`**, la libreria de **`Pandas`** el denominador lo utiliza **`(n - 1)`**, exactamente igual que en la imagen, siendo el **`1`** el **grado de libertad**. En caso contrario, la libreria de **`Numpy`**, utiliza como denominador **`(n)`**, dado que por defecto su **grado de libertad** es igual a **`0`**.

<center><img src="https://i.postimg.cc/05SSYZWt/dl5.png"></center>

Al dar un valor al grado de libertad, le estamos dando un grado mayor de libertad a las variables, dado que al tener un valor menor en el denominador tendremos un valor de la dispersión aún más grande de acuerdo al promedio, lo cual nos da un rango de libertad de accion para reaccionar segun el caso que estemos estudiando y sus datos. Mientras menor sea el grado de libertad menor tiempo de reaccion.

<center><img src="https://i.postimg.cc/Dwgb6kp2/dl6.png"></center>

In [9]:
df['peso'].var()

420.32222222222225

In [8]:
df['peso'].var(ddof=0)

378.29

In [ ]:
np.var(df['peso'])

378.29

#### **`Estadistica descriptiva - Calcular la DESVIACION ESTANDAR (DESVIACION TIPICA)`**

In [21]:
dt['passengers'].std()

119.96631694294321

Utiliando Numpy

In [22]:
np.std(dt['passengers'])

119.54904153277148

In [23]:
np.sqrt(np.var(dt['passengers']))

119.54904153277148